p2p/providers/peerstore_provider/persistence/p2p.go

```diff
@@ -0,0 +1,72 @@
+package persistence
+
+import (
+	"fmt"
+
+	"github.com/pokt-network/pocket/p2p/providers/peerstore_provider"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+)
+
+var (
+	_ peerstore_provider.PeerstoreProvider = &p2pPeerstoreProvider{}
+	_ p2pPStoreProviderFactory             = &p2pPeerstoreProvider{}
+)
+
+// unstakedPeerstoreProvider is an interface which the p2p module supports in
+// order to allow access to the unstaked-actor-router's peerstore
+//
+// TECHDEBT(#xxx): will become unnecessary after `modules.P2PModule#GetUnstakedPeerstore` is added.`
+// CONSIDERATION: split `PeerstoreProvider` into `StakedPeerstoreProvider` and `UnstakedPeerstoreProvider`.
+// (see: https://github.com/pokt-network/pocket/pull/804#issuecomment-1576531916)
+type unstakedPeerstoreProvider interface {
+	GetUnstakedPeerstore() (typesP2P.Peerstore, error)
+}
+
+type p2pPStoreProviderFactory = modules.Factory[peerstore_provider.PeerstoreProvider]
```

Comment:
Why not embed this in the struct as well?

Reply:
:100: nice simplification, thanks!

Reply:
So I know this instance is no longer applicable as I refactored this type out of existence in future commits but I'm noticing something that will continue to apply to the `persistencePeerstoreProvider` which was just me being confused initially.

As this is an alias to an interface type with the intention of being used for a compile-time check, it doesn't make sense for it to be embedded as that have the same effect. I had a conversation with chatGPT about this, here's the summary:

> Embedding an interface in a struct is a statement of intent that the struct should satisfy the interface, but it doesn't provide compile-time safety.
>
> The var _ SomeInterface = &myStruct{} check provides compile-time safety by ensuring that myStruct does indeed satisfy SomeInterface.